openprescribing/frontend/tests/commands/test_import_measures_new.py

```diff
@@ -190,15 +215,246 @@ def test_import_measures(self):
+        ]
+
+        prescribing_rows = []
+
+        for practice in Practice.objects.all():
+            for month in [7, 8]:
+                timestamp = '2018-0{}-01 00:00:00 UTC'.format(month)
+
+                for bnf_code, bnf_name in presentations:
+                    items = randint(0, 100)
+                    quantity = randint(6, 28) * items
+
+                    actual_cost = randint(100, 200) * quantity * 0.01
+
+                    # We don't care about net_cost.
+                    net_cost = actual_cost
+
+                    row = [
+                        'sha',  #  This value doesn't matter.
+                        practice.ccg.regional_team_id,
+                        practice.ccg.stp_id,
+                        practice.ccg_id,
+                        practice.code,
+                        bnf_code,
+                        bnf_name,
+                        items,
+                        net_cost,
+                        actual_cost,
+                        quantity,
+                        timestamp,
+                    ]
+
+                    prescribing_rows.append(row)
+
+        # In production, normalised_prescribing_standard is actually a view,
+        # but for the tests it's much easier to set it up as a normal table.
+        table = Client('hscic').get_or_create_table(
+            'normalised_prescribing_standard', schemas.PRESCRIBING_SCHEMA
+        )
+
+        # Upload prescribing_rows to normalised_prescribing_standard, and
+        # create a DataFrame for later verification.
+        with tempfile.NamedTemporaryFile() as f:
+            writer = csv.writer(f)
+            for row in prescribing_rows:
+                writer.writerow(row)
+            f.seek(0)
+            table.insert_rows_from_csv(f.name)
+
+            headers = [
+                'sha',
+                'regional_team_id',
+                'stp_id',
+                'ccg_id',
+                'practice_id',
+                'bnf_code',
+                'bnf_name',
+                'items',
+                'net_cost',
+                'actual_cost',
+                'quantity',
+                'month',
+            ]
+            prescriptions = pd.read_csv(f.name, names=headers)
+            prescriptions['month'] = prescriptions['month'].str[:10]
+
+        # Generate random practice statistics data.  This data is never saved
+        # to the database.
+        practice_statistics_rows = []
+        columns = [
+            'month',
+            'regional_team_id',
+            'stp_id',
+            'ccg_id',
+            'practice_id',
+            'total_list_size',
+        ]
+        practice_statistics = pd.DataFrame(columns=columns)
+
+        for practice in Practice.objects.all():
+            for month in [7, 8]:
+                timestamp = '2018-0{}-01 00:00:00 UTC'.format(month)
+
+                if month == 8 and practice.code == 'P00000':
```

Comment:
Comment explaining what's going on here?  Or perhaps an assertion like you do in the other test (`seen_practice_with_no_list_size = True` + `assert seen_practice_with_no_list_size`)?

Reply:
Good idea.